### PR TITLE
Prune unnecessary OCS resources

### PR DESCRIPTION
**Pull request by rexagod**
Prune unnecessary OCS resources that are not needed by the ManagedMCG
addon, namely:
* rook-ceph-operator
* ocs-operator
* ocs-metrics-exporter

### CODE REVIEW

**Comment by naveenpaul1** [comment]
@rexagod Danny mentioned that OB and OBC are dependent on OCS, 
Could you please check we are able to create OB and OBC even after downsizing OCS deployments.

**Comment by rexagod** [comment]
If we need OCS, then what's the point of all the Noobaa standalone effort that's happening? I made this PR with that perspective in mind, because if we are including OCS for OB/OBCs, then this PR makes no sense.

**Comment by naveenpaul1** [comment]
Yes @rexagod , That is something we need to discuss, I just wanted to know whether removing OCS related deployment will break our addon or not

**Comment by naveenpaul1** [comment]
// @v-harihar

**Comment by rexagod** [comment]
Closing this since we'll be removing ODF altogether, so no need to remove OCS components individually.

Will reopen if need be.

**Comment by rexagod** [comment]
Reopening, we might still need this (in case abstracting OB/OBCs does not work out).

File: controllers/managedmcg_controller.go [hunk 1]
```diff
@@ -319,6 +323,37 @@ func (r *ManagedMCGReconciler) reconcileNoobaaComponent() error {
...
+				*replicaCount = &zero
+				isChanged = true
+			}
+		default:
+			r.Log.Info("Could not find deployment", "Deployment", name)
+		}
+	}
+	if isChanged {
+		if err := r.update(csv); err != nil {
+			r.Log.Error(err, "Failed to update OCS CSV")
```
Comment (naveenpaul1) [inline controllers/managedmcg_controller.go:351]: do we need to return an error for this condition?

File: controllers/managedmcg_controller.go [hunk 2]
```diff
@@ -319,6 +323,37 @@ func (r *ManagedMCGReconciler) reconcileNoobaaComponent() error {
...
+	if csv, err = r.getCSVByPrefix("ocs-operator"); err != nil {
+		return err
+	}
+	var isChanged bool
+	deployments := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs
+	for i := range deployments {
+		name := deployments[i].Name
+		switch name {
+		case "ocs-operator", "ocs-metrics-exporter", "rook-ceph-operator":
+			zero := int32(0)
```
[the remark this answers is not in the document]
Comment (rexagod) [reply controllers/managedmcg_controller.go:338]: Can you elaborate why? I don't see any usages of this variable outside the loop.

File: controllers/managedmcg_controller.go (hunk 1 above)
Comment (rexagod) [reply controllers/managedmcg_controller.go:351]: Sounds good, adding.

File: controllers/managedmcg_controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (naveenpaul1) [reply controllers/managedmcg_controller.go:338]: code will run every time the loop executes, we don't need to create new variable and initialize it again and agin.

**Comment by naveenpaul1** [comment]
LGTM

**Pull request merged**